back/engines/commercial/user_custom_fields/app/controllers/user_custom_fields/web_api/v1/ref_distributions_controller.rb

```diff
@@ -41,18 +46,34 @@ def side_fx
 
       def reference_distribution
         @reference_distribution ||= authorize(
-          Representativeness::RefDistribution
-            .includes(:options)
-            .find_by!(custom_field_id: params[:custom_field_id])
+          RefDistribution.find_by!(custom_field_id: params[:custom_field_id])
         )
       end
 
       def create_params
-        params.permit(:custom_field_id, distribution: {})
+        params
+          .permit(:custom_field_id, distribution: {})
+          .merge(type: infer_ref_distribution_type)
+      end
+
+      def infer_ref_distribution_type
+        custom_field = CustomField.find(params[:custom_field_id])
+        if custom_field.key == 'birthyear'
+          'UserCustomFields::Representativeness::BinnedDistribution'
```

Comment:
Wouldn't it be better to write:
```suggestion
          UserCustomFields::Representativeness::BinnedDistribution.name
```
Same comment on line 64.